### PR TITLE
Added 6.5 support for upgrade

### PR DESCRIPTION
**Pull request by ntkathole**
Get ready for 6.5

### CODE REVIEW

**Comment by jyejare** [comment]
We are going to branch the 6.5 as master, and hence these changes are no more required. Feel free to close this PR.

**Comment by ntkathole** [comment]
@jyejare I think of merging this changes after branching... anyhow we need to do changes for 6.5 after branching... going onwards we can remove other versions if plan is to branch for each sat release.

**Comment by jyejare** [comment]
@ntkathole , Now we have branched 64less, so now we can remove all the code related 64 and less versions. 
We can assume TO_VERSION is 6.5 and FROM_VERSION is 6.4 and hence we can now remove all the loops regarding to and from versions.

**Comment by ntkathole** [comment]
For travis fix, raised issue #239

File: fabfile.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 from automation_tools import partition_disk
 
-from upgrade.runner import ( # flake8: noqa
+from upgrade.runner import (  # flake8: noqa
```
Comment (jyejare) [inline fabfile.py]:
```suggestion
from upgrade.runner import ( 
```

File: upgrade/capsule.py [hunk 2]
```diff
@@ -129,7 +129,7 @@ def satellite6_capsule_upgrade(cap_host, sat_host):
     # Host and Content Host are in different organizations (e.g. host not in
     # org, and content host in one), the content host will be unregistered as
     # part of the upgrade process.
-    if to_version in ['6.2', '6.3', '6.4']:
+    if to_version in ['6.2', '6.3', '6.4', '6.5']:
```
Comment (jyejare) [inline upgrade/capsule.py]:
```suggestion
    if float(to_version) >=6.2:
```
Not mandatory as we are going to remove this everything soon.

File: upgrade/capsule.py [hunk 3]
```diff
@@ -177,7 +177,7 @@ def satellite6_capsule_upgrade(cap_host, sat_host):
     if to_version == '6.1':
         run('capsule-installer --upgrade --certs-tar '
             '/home/{0}-certs.tar'.format(cap_host))
-    elif to_version == '6.3' or to_version == '6.4':
+    elif to_version in ['6.3', '6.4', '6.5']:
```
Comment (jyejare) [inline upgrade/capsule.py]: same as above

File: upgrade/client.py [hunk 4]
```diff
@@ -182,7 +182,10 @@ def satellite6_client_upgrade(os_version, clients, puppet=False):
     rhel_ver = os_version[-1]
     old_repo = 'rhel-{0}-server-satellite-tools-{1}-rpms'.format(
         rhel_ver, old_version)
-    puppet_agent = 'puppet' if to_version not in ['6.4'] else 'puppet-agent'
+    if to_version in ['6.1', '6.2', '6.3']:
```
Comment (jyejare) [inline upgrade/client.py]:
```suggestion
    puppet_agent = 'puppet' if float(to_version) <=6.3 else 'puppet-agent'
```

File: upgrade/helpers/tasks.py [hunk 5]
```diff
@@ -59,14 +59,14 @@ def check_necessary_env_variables_for_upgrade(product):
         failure.append('Product name should be one of {0}.'.format(
             ', '.join(products)))
     # From which version to upgrade
-    supported_vers = ['6.4', '6.3', '6.2', '6.1', '6.0']
+    supported_vers = ['6.5', '6.4', '6.3', '6.2', '6.1', '6.0']
```
Comment (jyejare) [inline upgrade/helpers/tasks.py]: Supported version is used for from versions I believe and hence we cannot have 6.5 there as of now.

File: upgrade/helpers/tasks.py [hunk 6]
```diff
@@ -59,14 +59,14 @@ def check_necessary_env_variables_for_upgrade(product):
...
     if os.environ.get('FROM_VERSION') not in supported_vers:
         failure.append('Wrong FROM_VERSION provided to upgrade from. '
-                       'Provide one of 6.4, 6.3, 6.2, 6.1, 6.0')
+                       'Provide one of 6.5, 6.4, 6.3, 6.2, 6.1, 6.0')
```
Comment (jyejare) [inline upgrade/helpers/tasks.py]:
```suggestion
                       'Provide one of {}'.format(supported_vers))
```

File: upgrade/helpers/tasks.py [hunk 7]
```diff
@@ -117,7 +117,7 @@ def sync_capsule_repos_to_upgrade(capsules):
     to_version = os.environ.get('TO_VERSION')
     os_ver = os.environ.get('OS')[-1]
     tools_repo_url = os.environ.get('TOOLS_URL_RHEL7') if to_version in [
-        '6.4', '6.3'] else None
+        '6.5'] else None
```
Comment (jyejare) [inline upgrade/helpers/tasks.py]: only 6.5 ??

File: upgrade/helpers/tasks.py [hunk 8]
```diff
@@ -156,7 +156,7 @@ def sync_capsule_repos_to_upgrade(capsules):
     except KeyError:
         # If latest capsule repo is not created already(Fresh Upgrade),
         # So create new....
-        if to_version in ['6.4', '6.3']:
+        if to_version in ['6.5', '6.4', '6.3']:
```
Comment (jyejare) [inline upgrade/helpers/tasks.py]: same as float related comment.

File: upgrade/helpers/tasks.py [hunk 9]
```diff
@@ -486,7 +486,7 @@ def capsule_sync(cap_host):
     :param list cap_host: List of capsules to perform sync
     """
     set_hammer_config()
-    if os.environ.get('TO_VERSION') in ['6.2', '6.3', '6.4']:
+    if os.environ.get('TO_VERSION') in ['6.2', '6.3', '6.4', '6.5']:
```
Comment (jyejare) [inline upgrade/helpers/tasks.py]: same as float related comment.

File: upgrade/helpers/tasks.py [hunk 10]
```diff
@@ -615,8 +615,7 @@ def upgrade_using_foreman_maintain():
                 ' -y'.format(os.environ.get('TO_VERSION')))
 
     # whitelist disk-performance check
-    # for 6.4 and 6.4.z upgrade.
-    if os.environ.get('TO_VERSION') in ['6.3', '6.4']:
+    if os.environ.get('TO_VERSION') in ['6.3', '6.4', '6.5']:
```
Comment (jyejare) [inline upgrade/helpers/tasks.py]: same as above

File: upgrade/helpers/tasks.py (hunk 5 above)
Comment (ntkathole) [reply upgrade/helpers/tasks.py]: Added 6.5 if someone wants to perform snap to snap upgrade.

**Comment by ntkathole** [comment]
Merging to trigger upgrade!

**Pull request merged**